straxen/plugins/event_area_per_channel.py

```diff
@@ -0,0 +1,38 @@
+import numpy as np
+import strax
+import straxen
+export, __all__ = strax.exporter()
+
+@export
+@strax.takes_config(
+    strax.Option('n_tpc_pmts', type=int,
+                 help='Number of TPC PMTs'),
+)
+class event_area_per_channel(strax.LoopPlugin):
+    """
+    Simple plugin that provides area per channel for main 
+    and alternative S1/S2 in the event. 
+    """
+    depends_on = ('events', 'event_basics', 'peaks')
+    parallel = False
+    provides = "event_area_per_channel"
+    __version__ = '0.0.0'
+    def infer_dtype(self):
+        dtype = []
+        dtype.append( (("Area per channel for main S2", "s2_area_per_channel"), 
+                       np.float32, (self.config['n_tpc_pmts'],)) )
+        dtype.append( (("Area per channel for alternative S2", "alt_s2_area_per_channel"), 
+                       np.float32, (self.config['n_tpc_pmts'],)) )
+        dtype.append( (("Area per channel for main S1", "s1_area_per_channel"), 
+                       np.float32, (self.config['n_tpc_pmts'],)) )
+        dtype.append( (("Area per channel for alternative S1", "alt_s1_area_per_channel"), 
+                       np.float32, (self.config['n_tpc_pmts'],)) )
+        dtype += strax.time_fields
+        return dtype
+    
+    def compute_loop(self, event, peaks):
+        result = dict()
+        result['time'], result['endtime'] = event['time'], strax.endtime(event)
+        for type_ in ['s1','s2','alt_s1','alt_s2']:
+            result[type_+'_area_per_channel'] = (peaks['area_per_channel'][event[type_+"_index"]] if event[type_+"_index"]!=-1 else 0.0)
+        return result
```

Comment:
```suggestion
    def compute_loop(self, event, peaks):
        result = np.zeros(1, self.dtype)
        result['time'] = event['time']
        result['endtime'] = strax.endtime(event)
        
        for type_ in ['s1', 's2', 'alt_s1', 'alt_s2']:
            type_index = event[f'{type_}_index']
            if type_index != -1:
                type_ara_per_channel = peaks['area_per_channel'][type_index]
                result[f'{type_}_area_per_channel'] = type_ara_per_channel
        return result
```

The 0.0 wouldn't have worked right?

Reply:
Right! Like this though, strax/straxen complains:
`AttributeError: Please provide result in compute loop as dict`
I have modified with the last push

Reply:
thanks!